OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -85,10 +85,16 @@ public override bool Tick(Actor self)
 					return false;
 
 				case PickupState.LockCarryable:
-					if (!carryable.LockForPickup(cargo, self))
+					var lockResponse = carryable.LockForPickup(cargo, self);
+					if (lockResponse == LockResponse.Failed)
 						Cancel(self);
 
-					state = PickupState.Pickup;
+					if (lockResponse == LockResponse.Success)
+						state = PickupState.Pickup;
+
+					if (lockResponse == LockResponse.Pending)
```

Comment:
Style nit: Can we please write this as if ... else if ... else? The Success and Pending checks are redundant if it failed, and this makes it clear that idling should be the default fallback if not otherwise handled.

Reply:
Fixed.